src/pages/Posts.js

```diff
@@ -4,27 +4,48 @@ import "../styles/Posts.css";
 import Card from "../components/Card";
 
 const Posts = () => {
-  const loggedInUser = JSON.parse(localStorage.getItem("selectedUser"));
   const [posts, setPosts] = React.useState([]);
+  const [allUsers, setAllUsers] = React.useState([]);
+  const allUsersAndPosts = [
+    {
+      name: "",
+      post: "",
+      title: "",
+    },
+  ];
 
   React.useEffect(function () {
     fetch("https://jsonplaceholder.typicode.com/posts")
       .then((res) => res.json())
       .then((data) => setPosts(data));
   }, []);
 
-  const selectedPost = posts.filter((item) => {
-    if (loggedInUser.id === item.userId) {
-      return item;
-    }
+  React.useEffect(function () {
+    fetch("https://jsonplaceholder.typicode.com/users")
+      .then((res) => res.json())
+      .then((data) => setAllUsers(data));
+  }, []);
+
+  allUsers.map((userItem) => {
+    posts.map((postItem) => {
+      if (userItem.id === postItem.userId) {
+        allUsersAndPosts.push({
+          name: userItem.name,
+          post: postItem.body,
+          title: postItem.title,
+        });
+      }
+    });
```

Comment:
Use a state to store the values in side a react component and never a const. const will get initialized every time your component re-renders.
 
Please user forEach here if you are not returning any values. 

https://www.freecodecamp.org/news/4-main-differences-between-foreach-and-map/